tests/ui/components/filters/sortoptionscomponent.js

```diff
@@ -182,4 +183,60 @@ describe('sort options component', () => {
     const wrapper = mount(component);
     expect(wrapper.text()).toEqual('');
   });
+
+  it('uses the persisted sortBys on load', () => {
+    const opts = {
+      ...defaultConfig,
+      options: threeOptions
+    };
+    const setSortBys = jest.fn();
+    COMPONENT_MANAGER.core.setSortBys = setSortBys;
+    COMPONENT_MANAGER.core.storage.setWithPersist(StorageKeys.SORT_BYS, [ threeOptions[1] ]);
+    const component = COMPONENT_MANAGER.create('SortOptions', opts);
+    const wrapper = mount(component);
+    expect(setSortBys).toHaveBeenCalledTimes(0);
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(2).getDOMNode().checked).toBeTruthy();
+  });
+
+  it('ignores persisted sortBys that do not match any options from config', () => {
+    const opts = {
+      ...defaultConfig,
+      options: threeOptions
+    };
+    COMPONENT_MANAGER.core.storage.setWithPersist(StorageKeys.SORT_BYS, [{
+      type: 'fake type!'
+    }]);
+    const component = COMPONENT_MANAGER.create('SortOptions', opts);
+    const wrapper = mount(component);
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(0).getDOMNode().checked).toBeTruthy();
+  });
+
+  it('sets persisted sort option on back navigation', () => {
+    const opts = {
+      ...defaultConfig,
+      options: threeOptions
+    };
+    const component = COMPONENT_MANAGER.create('SortOptions', opts);
+    const wrapper = mount(component);
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(0).getDOMNode().checked).toBeTruthy();
+    COMPONENT_MANAGER.core.storage.setWithPersist(StorageKeys.SORT_BYS, [threeOptions[2]]);
+    COMPONENT_MANAGER.core.storage.set(StorageKeys.HISTORY_POP_STATE, {});
+    wrapper.update();
+    expect(wrapper.find('.yxt-SortOptions-optionSelector').at(3).getDOMNode().checked).toBeTruthy();
```

Comment:
Why are we checking at index 3 if there are only three options in the array?

Reply:
sort options adds a default option even if you have no options (very weird I know, I think this was part of the original SortOptions spec)